code/modules/mob/living/basic/space_fauna/wumborian_fugu/inflation.dm

```diff
@@ -0,0 +1,92 @@
+/**
+ * Action which inflates you, making you larger and stronger for the duration. Also invulnerable.
+ * This is pretty much all just handled by a status effect.
+ * Unfortunately the requirements here are specific enough that it kind of only works for the mob it is designed for.
+ */
+/datum/action/cooldown/fugu_expand
+	name = "Inflate"
+	desc = "Temporarily increases your size, making you significantly more dangerous and durable!"
+	button_icon = 'icons/mob/actions/actions_animal.dmi'
+	button_icon_state = "expand"
+	background_icon_state = "bg_fugu"
+	overlay_icon_state = "bg_fugu_border"
+	cooldown_time = 16 SECONDS
+	melee_cooldown_time = 0 SECONDS
+
+/datum/action/cooldown/fugu_expand/IsAvailable(feedback)
+	. = ..()
+	if (!.)
+		return FALSE
+	if(!istype(owner, /mob/living/basic/wumborian_fugu)) // A shame but there's not any good way to make this work on other mobs
+		if (feedback)
+			owner.balloon_alert(owner, "not stretchy enough!")
+		return FALSE
+	if(HAS_TRAIT(owner, TRAIT_FUGU_GLANDED))
+		if (feedback)
+			owner.balloon_alert(owner, "ALREADY WUMBO!")
+		return FALSE
+	return TRUE
+
+/datum/action/cooldown/fugu_expand/Activate(atom/target)
+	. = ..()
+	var/mob/living/living_owner = owner
+	living_owner.apply_status_effect(/datum/status_effect/inflated)
+
+/**
+ * Status effect from the Expand action, makes you big and round and strong.
+ */
+/datum/status_effect/inflated
+	id = "wumbo_inflated"
+	duration = 10 SECONDS
+	alert_type = /atom/movable/screen/alert/status_effect/inflated
+
+/atom/movable/screen/alert/status_effect/inflated
+	name = "WUMBO"
+	desc = "You feel big and strong!"
+	icon_state = "gross"
+
+/datum/status_effect/inflated/on_creation(mob/living/new_owner, ...)
+	if (!istype(new_owner, /mob/living/basic/wumborian_fugu))
+		return FALSE
+	return ..()
+
+/datum/status_effect/inflated/on_apply()
+	. = ..()
+	RegisterSignal(owner, COMSIG_MOB_STATCHANGE, PROC_REF(check_death))
+	owner.add_movespeed_modifier(/datum/movespeed_modifier/status_effect/inflated)
+	ADD_TRAIT(owner, TRAIT_FUGU_GLANDED, TRAIT_STATUS_EFFECT(id))
+	owner.AddElement(/datum/element/wall_smasher)
+	owner.mob_size = MOB_SIZE_LARGE
+	owner.icon_state = "Fugu1"
+	owner.melee_damage_lower = 15
+	owner.melee_damage_upper = 20
+	owner.status_flags |= GODMODE
+	var/mob/living/basic/basic_owner = owner
+	basic_owner.obj_damage = 60
+	basic_owner.ai_controller.blackboard[BB_BASIC_MOB_FLEEING] = FALSE
+	basic_owner.ai_controller.CancelActions()
```

Comment:
Aren't there cases in which we could theoretically see this effect on a non-basic mob (i.e. someone in the future adds this as their cool new action)? If so, add a typecheck just in case we don't modify non-existent vars on a carbon or something

Reply:
I tried to figure out a way to enable those future cases but in this case the behaviour is _too_ specific.
There's already a type check, the ability simply fails to trigger if you're not the appropriate kind of mob when you use it.
Additionally, the status effect fails to apply to anything which isn't the correct type of mob as well.

I got as far as:
```
owner.add_movespeed_modifier(/datum/movespeed_modifier/status_effect/inflated)
ADD_TRAIT(owner, TRAIT_FUGU_GLANDED, TRAIT_STATUS_EFFECT(id))
owner.AddElement(/datum/element/wall_smasher)
owner.mob_size = MOB_SIZE_LARGE
```
but anything under that was just too specific to this one use case for me to have a sensible way of preserving the behaviour while also making it a sane option for any other mob.

Reply:
This is notably also true of the previous incarnation's ability, but that one for some reason had _half_ of the implementation inside an ability and then half in a callback on the mob? Reading it was what drove me to convert the mob in the first place.

Reply:
It was already type guarded for being a `/mob/living/basic` by the line in `on_creation` but this has made it clear to me that the _code_ doesn't describe that so I've rewritten it in a way that if anyone touches this in the future it will be transparent that this only works on one specific kind of mob and if they change that they will have to change other things.